tests/test_integration.py `test_jupyter_appnope_env_markers` fails with ipykernel 5.0.0 release

As seen in CI:
```
=================================== FAILURES ===================================
_______________________ test_jupyter_appnope_env_markers _______________________
    @pytest.mark.skipif(NOT_CPYTHON27)
    def test_jupyter_appnope_env_markers():
      res = run_pex_command(['--disable-cache',
                             'jupyter==1.0.0',
                             '-c', 'jupyter',
                             '--',
                             '--version'])
>     res.assert_success()
tests/test_integration.py:882: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = results(output=[], return_code=None, exception=Untranslateable("Package Source...able by ChainedTranslator(WheelTranslator, EggTranslator, SourceTranslator)\n')
    def assert_success(self):
      if not (self.exception is None and self.return_code in [None, 0]):
        raise AssertionError(
          'integration test failed: return_code=%s, exception=%r, output=%s, traceback=%s' % (
>           self.return_code, self.exception, self.output, self.traceback
          )
        )
E       AssertionError: integration test failed: return_code=None, exception=Untranslateable("Package SourcePackage(u'https://files.pythonhosted.org/packages/e1/cb/0129f461e8a4823fe662df4992d6ce8a0b593a5e86c6de78d4b882a0cf44/ipykernel-5.0.0.tar.gz#sha256=3e0ffdf545c0bf80d9dab6523ec6829831408c474772487aeb6eb9f0348b6a1e') is not translateable by ChainedTranslator(WheelTranslator, EggTranslator, SourceTranslator)",), output=[], traceback=Traceback (most recent call last):
E         File "/home/travis/build/pantsbuild/pex/.tox/py27-integration/lib/python2.7/site-packages/pex/testing.py", line 258, in run_pex_command
E           main(args=args)
E         File "/home/travis/build/pantsbuild/pex/.tox/py27-integration/lib/python2.7/site-packages/pex/bin/pex.py", line 737, in main
E           pex_builder = build_pex(reqs, options, resolver_options_builder)
E         File "/home/travis/build/pantsbuild/pex/.tox/py27-integration/lib/python2.7/site-packages/pex/bin/pex.py", line 667, in build_pex
E           for dist in resolveds:
E         File "/home/travis/build/pantsbuild/pex/.tox/py27-integration/lib/python2.7/site-packages/pex/resolver.py", line 527, in resolve_multi
E           use_manylinux=use_manylinux):
E         File "/home/travis/build/pantsbuild/pex/.tox/py27-integration/lib/python2.7/site-packages/pex/resolver.py", line 453, in resolve
E           return resolver.resolve(resolvables_from_iterable(requirements, builder))
E         File "/home/travis/build/pantsbuild/pex/.tox/py27-integration/lib/python2.7/site-packages/pex/resolver.py", line 248, in resolve
E           dist = self.build(package, resolvable.options)
E         File "/home/travis/build/pantsbuild/pex/.tox/py27-integration/lib/python2.7/site-packages/pex/resolver.py", line 206, in build
E           raise Untranslateable('Package %s is not translateable by %s' % (package, translator))
E       Untranslateable: Package SourcePackage(u'https://files.pythonhosted.org/packages/e1/cb/0129f461e8a4823fe662df4992d6ce8a0b593a5e86c6de78d4b882a0cf44/ipykernel-5.0.0.tar.gz#sha256=3e0ffdf545c0bf80d9dab6523ec6829831408c474772487aeb6eb9f0348b6a1e') is not translateable by ChainedTranslator(WheelTranslator, EggTranslator, SourceTranslator)
.tox/py27-integration/lib/python2.7/site-packages/pex/testing.py:229: AssertionError
========= 1 failed, 41 passed, 3 skipped, 1 xfailed in 344.62 seconds ==========
ERROR: InvocationError for command '/home/travis/build/pantsbuild/pex/.tox/py27-integration/bin/py.test tests/test_integration.py -vvs' (exited with code 1)
___________________________________ summary ____________________________________
ERROR:   py27-integration: commands failed
```

--- a/tests/test_integration.py
+++ b/tests/test_integration.py
@@ -874,8 +874,13 @@ def test_platform_specific_egg_resolution_matching():
 
 @pytest.mark.skipif(NOT_CPYTHON27)
 def test_jupyter_appnope_env_markers():
+  # NB: jupyter 1.0.0 does not properly pin its dep graph and both ipykernel and ipython support
+  # python 3 only post 5 and 6 respectively; so we add those deps explictly only to constrain them
+  # low such that they support python 2.7 for this test.
   res = run_pex_command(['--disable-cache',
                          'jupyter==1.0.0',
+                         'ipykernel<5',
+                         'ipython<6',
                          '-c', 'jupyter',
                          '--',
                          '--version'])

